src/client.ts

```diff
@@ -0,0 +1,153 @@
+import { Transform, TransformCallback } from 'node:stream';
+
+import fetchRetry from 'fetch-retry';
+import fetch from 'cross-fetch';
+
+import { InvalidInputError } from './errors.js';
+import { version } from './buildInfo.js';
+import { lookupApiKey, lookupEndpoint } from './helpers/config.js';
+import {
+  ApiClient,
+  ApiClientOptions,
+  ApiClientResponse,
+  createApiClient,
+} from './api/client.js';
+import { clientErrorWrapper } from './utils/errors.js';
+import { OmitVersion } from './utils/types.js';
+import { ApiEventClient, createApiEventClient } from './api/event-client.js';
+
+export type RawHeaders = Record<string, string>;
+
+export interface Configuration {
+  apiKey?: string;
+  endpoint?: string;
+  headers?: RawHeaders;
+}
+
+export type Options = { signal?: AbortSignal };
+
+export class Client {
+  readonly #client: ApiClient;
+  readonly #eventClient: ApiEventClient;
```

Comment:
I would make these two `protected` with `_` prefix, this ensures that one can use them in case of extending our client via inheritance.